plugins/openai/newhelm/annotators/openai_compliance_annotator.py

```diff
@@ -0,0 +1,113 @@
+import re
+from typing import Callable, Dict, List, Optional
+from pydantic import BaseModel
+
+from newhelm.base_annotator import BaseAnnotator
+from newhelm.single_turn_prompt_response import PromptInteraction
+from newhelm.sut import SUTResponse, PromptResponseSUT
+from newhelm.prompt import Prompt, SUTOptions
+from newhelm.suts.openai_client import OpenAIChat
+
+
+class ComplianceAnnotation(BaseModel):
+    """Annotates level of SUT compliance (Full compliance, full refusal, or partial refusal)"""
+
+    class Completion(BaseModel):
+        compliance: str
+
+    class Interaction(BaseModel):
+        completions: List["ComplianceAnnotation.Completion"]
+
+    interactions: List[Interaction]
+
+
+class OpenAIComplianceAnnotator(BaseAnnotator[ComplianceAnnotation]):
+    """Send the SUT's completions to gpt-3.5-turbo to get compliance classification."""
+
+    def __init__(
+        self,
+        formatter: Optional[Callable[[str, str], str]] = None,
+        decoder: Optional[Dict[str, str]] = None,
+        sut: Optional[PromptResponseSUT] = None,
+    ):
+        self.model = OpenAIChat(_MODEL_NAME) if sut is None else sut
+        self.formatter = _default_formatter if formatter is None else formatter
+        self.decoder = _DEFAULT_MAPPING if decoder is None else decoder
+
+    def annotate_test_item(
+        self, interactions: List[PromptInteraction]
+    ) -> ComplianceAnnotation:
+        """Returns an annotation for a single TestItem's interactions."""
+        openai_interactions = []
+        for interaction in interactions:
+            openai_completions = []
+            for completion in interaction.response.completions:
+                openai_prompt = Prompt(
+                    text=self.formatter(
+                        interaction.prompt.prompt.text, completion.text
+                    ),
+                    options=SUTOptions(temperature=0),
+                )
+                request = self.model.translate_request(openai_prompt)
```

Comment:
This should be `translate_text_prompt` as of #160.

Reply:
Done